File: LeetCode/cpp/415.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        int m = num1.size(), n = num2.size();
        string longer, shorter;
        if (m >= n) {
            longer = num1;
            shorter = num2;
        } else {
            longer = num2;
            shorter = num1;
            swap(m, n);
        }
        int carry = 0, i = 1;
        while (i <= n) {
            int sum = longer[m - i] - '0' + shorter[n - i] - '0' + carry;
            carry = sum / 10;
            sum %= 10;
            longer[m - i] = sum + '0';
            ++i;
        }
        while (carry) {
            if (m - i < 0) {
                longer.insert(longer.begin(), '1');
                break;
            }
            int sum = longer[m - i] - '0' + carry;
            carry = sum / 10;
            sum %= 10;
            longer[m - i] = sum + '0';
            ++i;
        }
        return longer;
    }
};

File: LeetCode/python/415.py
class Solution(object):
    def addStrings(self, num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        m, n = len(num1), len(num2)
        longer, shorter = list(num1), list(num2)
        if m < n:
            longer, shorter = shorter, longer
            m, n = n, m
        i, carry = 1, 0
        while n - i >= 0:
            sum = ord(longer[m - i]) - ord('0') + ord(shorter[n - i]) - ord('0') + carry
            carry = sum // 10
            sum %= 10
            longer[m - i] = str(sum)
            i += 1
        while carry:
            if m - i < 0:
                longer = ['1'] + longer
                break
            sum = ord(longer[m - i]) - ord('0') + carry
            carry = sum // 10
            sum %= 10
            longer[m - i] = str(sum)
            i += 1
        return "".join(longer)
